src/systems/filecoin_nodes/clock/clock_subsystem.go

```diff
@@ -2,10 +2,27 @@ package clock
 
 import "time"
 
-func (_ *WallClock_I) NowUTC() Time {
+func (_ *Clock_I) UTCTime() Time {
 	return Time(time.Now().Unix())
 }
 
-func (_ *WallClock_I) NowUTCNano() Time {
+func (_ *Clock_I) UnixNano() Time {
 	return Time(time.Now().UnixNano())
 }
+
+func (self *Clock_I) CurrentEpoch() ChainEpoch {
+	return self.currentEpoch_
+}
```

Comment:
- This should not live here. the `Clock` does not know the epoch, the Blockchain does, and that's related to finality in chainsync, bootstrap state and more. `Clock` here should not have (nor report) the epoch -- defer to the blockchain.

- there was a previous idea of a `Round/EpochClock` but that was going to be a smalll abstraction on top of (having a pointer to) the blockchain system, and deferring to it for calculation (including failing to return because system is not in a state able to return an epoch). we ended up not need it so i didnt write it.

Reply:
Oh maybe you want to calculate what epoch _it should be_ based on a prior epoch-to-time mapping? I think functions like this are appropriate for this package:

```go
func EpochAfterTime(start block.ChainEpoch, t Time) block.ChainEpoch {
   // returns start + EpochsInSeconds(t)
}

// genesis being hard coded
func EpochAfterGenesis() block.ChainEpoch {
   // returns EpochAfterTime(genesis, Now() - timeOfGenesis)
}
```

then caller can take responsibility of passing in the right start epoch, etc.

Reply:
epochDuration (or EpochsInSeconds) should be pulled from blockchain?